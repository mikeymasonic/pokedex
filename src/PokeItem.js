import React, { Component } from "react";
import PokeList from "./PokeList.js";
// import pokeData from "./data.js";
import request from 'superagent';

export default class Body extends Component {

    state = {
        data: [],
        selected: null,
        };

        async componentDidMount() {
            const data = await request.get(`https://alchemy-pokedex.herokuapp.com/api/pokedex`);
            this.setState({ data: data.body.results })         
        }

  render() {
    console.log(this.state.selected);
    const pokeAPIData = this.state.data
    const pokeNodes = pokeAPIData
      .filter(poke => {
        if (!this.state.selected) return true;

        return poke.pokemon === this.state.selected || String(poke.hp) === this.state.selected || String(poke.speed) === this.state.selected;
      })
      .map((poke) => <PokeList key={poke._id} poke={poke} />);
    
    // let unique = [...new Set(pokeData.keyword)];
    // console.log(unique);
    const handleChange = e => {
      this.setState({ selected: e.target.value });
    };
    return <div> 
       <main>
        <section className="options">
            <select className="poke-name-filter" onChange={handleChange}>
              <option value="" defaultValue>
                Pokemon
              </option>
              
              {pokeAPIData.map(poke => <option key={poke._id}>{poke.pokemon}</option>)};
            </select>

            <select className="poke-hp-filter" onChange={handleChange}>
              <option value="" defaultValue>
                HP
              </option>
              {pokeAPIData.map(poke => <option key={poke._id}>{poke.hp}</option>)};
            </select>

            <select className="poke-speed-filter" onChange={handleChange}>
              <option value="" defaultValue>
                Speed
              </option>
              {pokeAPIData.map(poke => <option key={poke._id}>{poke.speed}</option>)};
            </select>  
          </section>

          <section className="list-section">
            <ul className="pokes">{pokeNodes}</ul>
          </section>
        </main>
        
            </div>;
  }
}

//this is my deck - how you want it to run