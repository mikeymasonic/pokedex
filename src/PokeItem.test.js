import React from 'react';
import PokeItem from './PokeItem.js';
import renderer from "react-test-renderer";

test('renders PokeItem.js correctly', () => {
  const tree = renderer
  .create(<PokeItem />)
  .toJSON();
  expect(tree).toMatchSnapshot();
});