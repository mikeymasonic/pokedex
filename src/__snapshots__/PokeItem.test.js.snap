// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders PokeItem.js correctly 1`] = `
<div>
  <main>
    <section
      className="options"
    >
      <select
        className="poke-name-filter"
        onChange={[Function]}
      >
        <option
          defaultValue={true}
          value=""
        >
          Pokemon
        </option>
        ;
      </select>
      <select
        className="poke-hp-filter"
        onChange={[Function]}
      >
        <option
          defaultValue={true}
          value=""
        >
          HP
        </option>
        ;
      </select>
      <select
        className="poke-speed-filter"
        onChange={[Function]}
      >
        <option
          defaultValue={true}
          value=""
        >
          Speed
        </option>
        ;
      </select>
    </section>
    <section
      className="list-section"
    >
      <ul
        className="pokes"
      />
    </section>
  </main>
</div>
`;
